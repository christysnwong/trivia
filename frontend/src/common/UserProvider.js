import React, {useEffect, useState} from "react";
import UserContext from "./UserContext";

const demoUser = {
  id: 1,
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "test@test.com",
  stats: {
    level: 14,
    title: "Pro",
    "quizzes completed": 34,
    points: 3440,
    remainingPts: 60,
    levelPts: 300,
  },
};

const UserProvider = ({
  children,
  currUser = demoUser
}) => {
  
  const [currentUser, setCurrUser] = useState();

  return (
    // <UserContext.Provider value={{ currUser }}>
    <UserContext.Provider value={{ currUser, setCurrUser }}>
      {children}
    </UserContext.Provider>
  );
};

export default UserProvider;
