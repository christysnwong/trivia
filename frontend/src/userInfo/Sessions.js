import { useState, useEffect, useContext } from "react";
import TriviaApi from "../api/api";
import UserContext from "../common/UserContext";

import Table from "../userTable/Table";

const Sessions = () => {
  const [sessions, setSessions] = useState([]);
  const { currUser } = useContext(UserContext);

  // get played session history
  useEffect(() => {
    TriviaApi.getSessions(currUser.username).then((sessions) => {
      setSessions(sessions);
    });
  }, []);

  return (
    <div className="col-10 offset-1 col-lg-8 offset-lg-2 col-xxl-6 offset-xxl-3 col-xxxl-4 offset-xxxl-4">
      <h3 className="mb-5">
        <i className="fa-solid fa-clipboard me-2"></i>
        Recent Played Sessions
      </h3>

      {!sessions.length &&
        "There is no session data in this section. Try out a trivia quiz!"}

      {sessions.length > 0 && (
        <>
          <p className="mb-4">
            The following table shows your recent played sessions.
          </p>

          <Table entries={sessions} />
        </>
      )}
    </div>
  );
};

export default Sessions;

//
