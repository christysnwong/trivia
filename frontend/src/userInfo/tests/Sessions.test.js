import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import Sessions from "../Sessions";
import { MemoryRouter } from "react-router-dom";
import UserProvider from "../../common/UserProvider";

import TriviaApi from "../../api/api";

const { commonBeforeEach, commonAfterAll } = require("./_testCommonUserInfo");

beforeEach(commonBeforeEach);
afterAll(commonAfterAll);

it("renders without crashing", function () {
  render(
    <UserProvider>
      <Sessions />
    </UserProvider>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Sessions />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("checks if TriviaApi.getSessions is called and displayed properly", async () => {
  jest.spyOn(React, "useEffect").mockImplementation((f) => f());
  jest.spyOn(TriviaApi, "getSessions");
  render(
    <UserProvider>
      <Sessions />
    </UserProvider>
  );

  expect(TriviaApi.getSessions).toHaveBeenCalled();

  expect(await screen.findByText(/category/i)).toBeInTheDocument();
  expect(await screen.findByText(/difficulty/i)).toBeInTheDocument();

  const entries = await screen.findAllByText(/entertainment/i);
  expect(entries).toHaveLength(7); 


});