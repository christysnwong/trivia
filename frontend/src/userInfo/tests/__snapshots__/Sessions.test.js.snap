// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-10 offset-1 col-lg-8 offset-lg-2 col-xxl-6 offset-xxl-3 col-xxxl-4 offset-xxxl-4"
  >
    <h3
      class="mb-5"
    >
      <i
        class="fa-solid fa-clipboard me-2"
      />
      Recent Played Sessions
    </h3>
    There is no session data in this section. Try out a trivia quiz!
  </div>
</DocumentFragment>
`;
